### ISSUE TITLE
[skip ci] at Pull Request Level

### ISSUE THREAD

**Issue by jmendiara** [buildkite/feedback#432]
It would be nice to include support [skip ci] at Pull Request level.

This will help on PR workflows[1] like this one:
* Devs start a feature, and make commits
* Once they are done, they open a PR
* CI should test the PR, for example: Lint, unit test, build, publish some artifacts, deploy a transient environment, run small and fast e2e test suite, tear down environments... whatever

But... Sometimes devs are looking just for some feedback, making a WIP (work in progress) pull request, or they are updating some doc that will not change binaries...

Running a pipeline for this cases is money wasting, in terms of CI resources and developer time.

The current alternatives is squash & rebase, ammending the message to include [skip ci] at commit level. But manual rebasing is hard sometimes and github can make this for you when merging.

Analyzing [skip ci] labels in PRs, or in PR title/description as you do with commits, will make CI run smoother, saving some $$

[1] You can make a Pull Request only pipeline by limiting to build a very strange branch 
<img width="878" alt="captura de pantalla 2018-07-21 a las 8 33 56" src="https://user-images.githubusercontent.com/851359/43032890-eb813314-8cc0-11e8-9c4c-d0f19dbe521d.png">

**Comment by petemounce** [comment]
How about for branches like this, naming them `wip/some-name-here`, and then setting up buildkite with `!wip/*` so it doesn't build such branches or PRs?

**Comment by jmendiara** [comment]
@petemounce That could be a workaround, but it requires another contract in the way of working for devs that will require to push to another remote branch when done.

It's easier to add/remove labels, and pass the CI when no-label is present.

**Comment by petemounce** [comment]
That's true. It's quite a provider-specific way to do it, though - each provider would need to have some labelling mechanism, and each integration would need to grow to support it.

**Comment by davehowell** [comment]
> But... Sometimes devs are looking just for some feedback, making a WIP (work in progress) pull request, or they are updating some doc that will not change binaries...

For those specific scenarios, you could just make it not trigger a build if the PR has the "Draft" or "WIP" setting on it. Github, Gitlab and Bitbucket all have that option natively.

**Comment by chloeruka** [comment]
Thanks @davehowell 

We do have an upcoming conditional webhook feature that should give pipeline maintainers extra control over how and when builds are made. Keep an eye on our changelog. 😃
